### DISCUSSION TITLE
Error serializing - object Promise

### DISCUSSION THREAD

**Discussion by AndreHaupt85** [neuefische/questions#185] [Web] [answered]
Guten morgen,

ich habe beim rendern von von Daten beim dynamische Routing folgende Fehlermeldung die ich leider auch durch googlen nicht richtig zuordnen kann. Vielleicht kann mir da jemand helfen oder hatte die schon mal.

![Bildschirmfoto 2022-09-29 um 08 10 41](https://user-images.githubusercontent.com/109967968/192953206-16fabfa4-ec41-4d19-b46c-822c4456a820.png)

![Bildschirmfoto 2022-09-29 um 08 10 45](https://user-images.githubusercontent.com/109967968/192953223-aae6696a-668e-4a58-baac-965dbea4adb5.png)

![Bildschirmfoto 2022-09-29 um 08 37 33](https://user-images.githubusercontent.com/109967968/192957309-7a9a458e-29cd-466c-9c8a-3d2255dad8b4.png)

**Comment by actyralf** [comment]
I think line 21 should be `Recipe.findById(id)` instead of `recipe.findById(id)`.
Line 23 should then be `const { name } = recipes` instead of `const { name } = recipe`.

**Reply by AndreHaupt85** [reply]
that was both correct . thank u :)

it didn´t solve the Problem. but i also saw i forgot to send the 3. screenshot

**Reply by actyralf** [reply]
But even then I think your function `getRecipesById(id)` does not do what it's supposed to do: it will return a string (`name`) and not a recipe or a list of recipes as it's name suggests.

I think what you want to do is this: 

```
export async function getRecipeById(id){
   const recipe = await Recipe.findById(id);
   return recipe;
}
```

And don't forget to add `try/catch` whenever you call async database functions or http requests: they can always fail for external reasons (server down, no internet etc.) and your app should be able to handle this properly.

**Comment by AndreHaupt85** [comment, marked as answer]
Git the solutions with the coaches.

the problem was that the "ID" string was too complex to serialize.
